checkout/src/test/java/com/checkout/base/model/CardSchemeTest.kt

```diff
@@ -0,0 +1,54 @@
+package com.checkout.base.model
+
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+
+internal class CardSchemeTest {
+    @Test
+    fun `given visa scheme string value returns correct result`() {
+        // Given
+        val expectedCardScheme = CardScheme.VISA
+
+        // When
+        val actualCardScheme = CardScheme.fetchCardScheme("Visa")
+
+        // Then
+        Assertions.assertEquals(expectedCardScheme, actualCardScheme)
+    }
+
+    @Test
+    fun `given Diners scheme string value returns correct result`() {
+        // Given
+        val expectedCardScheme = CardScheme.DINERS_CLUB
+
+        // When
+        val actualCardScheme = CardScheme.fetchCardScheme("Diners Club International")
+
+        // Then
+        Assertions.assertEquals(expectedCardScheme, actualCardScheme)
+    }
+
+    @Test
+    fun `given American express scheme string value returns correct result`() {
+        // Given
+        val expectedCardScheme = CardScheme.AMERICAN_EXPRESS
+
+        // When
+        val actualCardScheme = CardScheme.fetchCardScheme("AMERICAN EXPRESS")
+
+        // Then
+        Assertions.assertEquals(expectedCardScheme, actualCardScheme)
+    }
```

Comment:
@okhan-okbay-cko This might help for the upper case. However, the scheme should be fetched for below string values:
1. Upper case
2. Lower case
3. Lower and upper case
4. Scheme string value with an underscore. For instance, `DINERS_CLUB_INTERNATIONAL`